### PR TITLE
fix(warnings): fixed warning when accessing context property from Request and Reply objects

### LINKED ISSUE #5033: reply.context.config is marked as deprecated in typescript but documented
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.23.0

### Plugin version

_No response_

### Node.js version

18.x

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

Mac

### Description

According to the docs: https://fastify.dev/docs/latest/Reference/Routes#config

One should access the request config object via `reply.context.config` but this is marked as deprecated in the typescript types.

```
/**
 * Route context object. Properties defined here will be available in the route's handler
 */
export interface FastifyContext<ContextConfig = ContextConfigDefault> {
  /**
   * @deprecated Use Request#routeConfig or Request#routeSchema instead
   */
  config: FastifyContextConfig & FastifyRouteConfig & ContextConfig;
}
```

### Steps to Reproduce

Try to add a response hook which accesses `reply.context.config`

```
fastify.addHook('onResponse', async (request, reply) => {
    if (reply.context.config['metrics']) {
      const status = reply.statusCode < 400 ? 'success' : 'error';
      logMetric(getRouteKey(request, status), reply);
    }
  });
```

### Expected Behavior

_No response_

### PR DESCRIPTION
**Pull request by giuliowaitforitdavide**
Closes #5033 

#### Checklist

- [X] run `npm run test` and `npm run benchmark`
- [X] tests and/or benchmarks are included
- [X] documentation is changed or added
- [X] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Review by metcoder95 — APPROVED** [review]
Nice one 🚀

File: docs/Reference/Request.md [hunk 1]
```diff
@@ -37,8 +37,8 @@ Request is a core Fastify object containing the following fields:
 - `connection` - Deprecated, use `socket` instead. The underlying connection of
   the incoming request.
 - `socket` - the underlying connection of the incoming request
-- `context` - A Fastify internal object. You should not use it directly or
-  modify it. It is useful to access one special key:	
+- `context` - Deprecated, a Fastify internal object. You should not use
```
[the remark this answers is not in the document]
Comment (giuliowaitforitdavide) [reply docs/Reference/Request.md:40]: You're right

**Comment by Uzlopak** [comment]
To be honest... I got super confused in the last 2-3 PR which are touching this code. So I refrain from reviewing :D.

**Comment by giuliowaitforitdavide** [comment]
Should we merge it?

**Pull request merged**